From patchwork Tue Sep  6 19:48:54 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968081
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 2CE2AC54EE9
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 19:49:47 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 6CE238D0009; Tue,  6 Sep 2022 15:49:45 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 6561D80008; Tue,  6 Sep 2022 15:49:45 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 459628D000D; Tue,  6 Sep 2022 15:49:45 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0011.hostedemail.com
 [216.40.44.11])
	by kanga.kvack.org (Postfix) with ESMTP id 2B2E78D0009
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 15:49:45 -0400 (EDT)
Received: from smtpin19.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay08.hostedemail.com (Postfix) with ESMTP id EA59E140876
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:44 +0000 (UTC)
X-FDA: 79882700688.19.12C6FBE
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf04.hostedemail.com (Postfix) with ESMTP id 69C7240071
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:44 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdbYx029984;
	Tue, 6 Sep 2022 19:49:41 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=B8vLfVMHYyq9eMWnxQSuV/JdLNr2sKQ59ECjk8k/jxg=;
 b=cF9Z0SJB39I3oWeDSd3vhASGKMwajfIWGQCZere6s4ZKIpNDANthfRGTBGHb0GBJsGQv
 wyjC2FF2PvSiYpJE5J9ZySG/I8atuI4HVAuwrSPr1E8iJ9yzGDfLIN3ejCd/nmPnSgSV
 z8laq0VjeFFpfeLoU6KB45+Zf2+0i2Y1/LQ0oil38LLLlNK/mRZejOOdbfZnCwSyB8q0
 arsEavbMRADdvBpCIAxrUiFSOE3rStMdFo7Gvw5MNQx6m5aOLREtCGxoEou11ghmjoWK
 OVaCz3m3Cg9wiShWe+wN8jZiU2wOrUW/dPXmWxYy3tCEg4ffSHW7xnVHguh3YksEGiz0 nQ==
Received: from iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta03.appoci.oracle.com [130.35.103.27])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbyftpvcv-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:41 +0000
Received: from pps.filterd
 (iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286Je5WO039711;
	Tue, 6 Sep 2022 19:49:40 GMT
Received: from nam10-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam10lp2108.outbound.protection.outlook.com [104.47.58.108])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc9m7j9-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:39 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=DHy5Yty+XNUCUei+XQiDIK86v74pZLMvTktMJM8dcJP7/4Puz5UitMqdFymJbHyunfP7hScZDorp2M+cH3ai9GDPf+2xofWIYptiW5P556AbFrPJwvWV4RKcw8MAyuuQmgFlre2d2fvJsT91R56On6lzueq2XTkn9mUdAETIfpR4LmUQUNyF31DgpMglqwTgxgcxigEYgZnoBjKvc380N6+sWG5vbq+Kdrw8AvZtOE6iCKwdj6ebDJINIVPxSVl8lX6kiSvMwe7eYy1tlz4+014xQ9oHvAni+1JgeU5SmDiIEwDXEdtAQgnzfG3qSgqj8cYgPrWiYG81QId9CFzpPw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=B8vLfVMHYyq9eMWnxQSuV/JdLNr2sKQ59ECjk8k/jxg=;
 b=gvIASdFIdAkiqitMO1LP6BWNZ5VD46nyQcyhoF0hDMJ8BLKK3sSUbWZbf2iV9X+1DvtI9clgDMloIhXYGhpLDcWAIJbqcT04VoIgWwc5rMyYHk/DYelR7S6w5sTibSzPwsAnfVcVNqRSzgu22BzXUO6YxbT2qJQX2VaDx43ygm+ImN6iodMRmmgvieQDJ7jKvtSL283HNJhqnMyoEXcAPA6cgtunqiv3pzCqSUyLfgqx/5Xu5JFfVjNeY64A6c50n0PXXJR9yZ9/IINtWhZbwIZN2BNRr/kWYwnwk+uZgh4+taGQxplbabxoUOAIYGrw0bq1jUTSBej6ldZgpcPi8A==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=B8vLfVMHYyq9eMWnxQSuV/JdLNr2sKQ59ECjk8k/jxg=;
 b=ZANBp2h5uANJ6aJFTsmtZA17uqBGU798Do4xPcQs40sdkpw/F39MyegJ/vD7tWocdS46iTB9tmNyodRUyAKAnrauQ0v1kA9+65+lmOyFMtPnKjAteh4760MzcHr0tBHPp1ZucL3Tnas388AI06DuyQi5/QLpGSaqAEXjze2jSG0=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by SJ0PR10MB5598.namprd10.prod.outlook.com (2603:10b6:a03:3d9::18) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.10; Tue, 6 Sep
 2022 19:49:37 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:49:37 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: "Matthew Wilcox (Oracle)" <willy@infradead.org>,
        Liam Howlett
	<liam.howlett@oracle.com>,
        Vlastimil Babka <vbabka@suse.cz>, Davidlohr Bueso
	<dave@stgolabs.net>
Subject: [PATCH v14 35/70] x86: remove vma linked list walks
Thread-Topic: [PATCH v14 35/70] x86: remove vma linked list walks
Thread-Index: AQHYwimypKb0MUGWG0K8mHUGpo+PcQ==
Date: Tue, 6 Sep 2022 19:48:54 +0000
Message-ID: <20220906194824.2110408-36-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 5dc4d55d-bd55-49e4-e06a-08da9040ee20
x-ms-traffictypediagnostic: SJ0PR10MB5598:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 s36OAyknJCwktCJmM8qhocHdQXBeOgPdoEAoSzdKrbuckoa9G/xDcqOOCf8+7+IoQX1MHzmFa/onjp00NYNXgqnbQYwq6Ewnk5lPS7ox2QLcaU3mgpCMPk2ydpDm2f/fiVEWZEwrAE+wSo8FNQbZMVxrZxTca/4/lyK8W8fWXlyA8qcdarynBxiPnvvZAcyN7wsUyVOIJfTO3cwzAGqcESG+zZYNbPLQM0n1jPu74sNS4HpGPRhdjvCvu0NKN25UEzHgD1Xg+t/+Bwuz3qpMJ6i2RMhwVyS0THWgK5Z+8mmA3LNhI5sbk0w4Jj9qiRp+olviJtTE8lv1lJeuRkxu03ijX1dtHLpxwXxzM3fbzmZmHs0xMt9t2bOK/I0sm0z1ZLXFu1gL/9C0i6lBPOoPVSfl8qPZTSGjb4MyrbtIb2P7uC61dJdKAg6dNV4fSqWVYaVsJBn6cVcmW/4aLqEnT3KnmfNupUVPm/T1UcbLRd9XS+BY7gaQIMpU22q7J4k/07CxmpaeyG668ey8PRcFs44A9cDupHf9cDgAxJ6PcDzTUuJE5SViy+WnYMHPiHBuUKgI76LsD4TSXg8Ej1yggUdfecpQs2gLnaTU3ntwG5Ku3lsIkLLsY+pq7a212yuG4g2ML16OUyviLmC7nTCpABQ6aroFHCsX7aPWjDumWNVqPBsFK7912ScQORSwM4ODVFuWjOsnH3OaB0CoDNUqIGAv2JxzJ4qxHd06XKx5MKhJn6EjbHp2S2JPl8LK/AScfKshG7ivU82P0OKpyANJgw==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(346002)(39860400002)(396003)(136003)(376002)(366004)(64756008)(66556008)(41300700001)(91956017)(76116006)(66476007)(66446008)(66946007)(71200400001)(54906003)(110136005)(36756003)(316002)(6512007)(44832011)(8676002)(122000001)(2906002)(6486002)(4326008)(186003)(8936002)(5660300002)(478600001)(38070700005)(38100700002)(2616005)(86362001)(26005)(1076003)(6666004)(6506007)(83380400001);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?b0Sb5EFiCkpTcZa6sTWo/EA?=
	=?iso-8859-1?q?uKEV/z7q5K+ownc/gbDOl0hR1mWWD5q/MJB/llYtDb/BlDm1xeavGDZGU3rg?=
	=?iso-8859-1?q?jnvb4Q2kxgbDhvX/iosKkKQU7lopnwfiJ4SgTD1N1G5huE0SVap3rB+d+qm+?=
	=?iso-8859-1?q?my5gdlSjDdROjqxKQz7DN9J08T4QJYGbm3X5QSb7koKDxEhRpbPluwMpjnCB?=
	=?iso-8859-1?q?t/WJ6dA3rgfHTqq4BB/Kx7nBv6HNOWtrWniv6Lka6W6BaB6ic6o5OofMHvdC?=
	=?iso-8859-1?q?OTdF+GqfgBJ10Rkj1nOX+mKZz6kx8VCOeK+QLKqtnB98Av+UIBmP8yhhf5Wc?=
	=?iso-8859-1?q?wcO0OX1RJ+2o+UYhqJLNoHoyWS04zAueRDGVMT0UBb3h6OsmN3AKbSXec4Jl?=
	=?iso-8859-1?q?8m+Bn2uIUkQnwfK4AzpvW4q/03+fPxEyOrPkh09IBIeAWy+5s+too4Xi+8iD?=
	=?iso-8859-1?q?h/aBbb0ncvj4EMNSmpWY52Iy7QlKptbnDIq1LaZf0qk3Zu1uY4S1iUxUcS+N?=
	=?iso-8859-1?q?Tbsu8vmRrjoxsUfXyfZsLR5GVRHCcINCq5nOvxebThKb78myl5dnkkhxtHgG?=
	=?iso-8859-1?q?c0o9xk8PNi807Ju/E/8ifVESwkKi2DS4kAEv9GZITD74kY5Ck8pw66PlP8q3?=
	=?iso-8859-1?q?fWz6SHDoYt0ShutuRTtdYtYodVS2FqK6nyRZkq/ExJaxQ57QfnAxxxJeJj3n?=
	=?iso-8859-1?q?72IQjdxaPTOLc322CaRaZufv3AEmwvYtTy4ggIvpLF8YX0ZbIqLVqOn/pzYh?=
	=?iso-8859-1?q?b3H2lrPY7pkFtUVv5aa6PwTwOVHPu8Sh1UqVwrOXWfU+fi/T+U8PH9rt6MFJ?=
	=?iso-8859-1?q?ZlHolF/mdCFDsPC6jLSRDfk7L7z+GsG0zqXSVbqROWKCWJET4A3BWtUZQw90?=
	=?iso-8859-1?q?nfxiMx4dCitHLmJK/75XJ3pcDMnBcBgg04MPfqFdoYOjyLc1/pdMnhYMyCk4?=
	=?iso-8859-1?q?m8kYENg+gormE65TixjAvAml6q1EhTwWJ1NKGN7FApM2ZWie9HAPgjaSN44l?=
	=?iso-8859-1?q?P2nFNxP+SJ4Gs5x3qUB8KMr6rIS2U6Xb2aoaDuoF4oK6pWDc+SzddxofveJG?=
	=?iso-8859-1?q?Ruu50BHohx+qfi0Xg40gGPeIO27TWlJOxoKuPzQkreCAYmuZ45BRcrVXNVEr?=
	=?iso-8859-1?q?BcHvyxWHZXKV0qYyEYw8RkGnNdg//8enSXa1zLE4LQLm66Nx9qG1BuWfpd0B?=
	=?iso-8859-1?q?BXcLSbA5esZhzf7RqqohKufrQWqLZsulAOAByBZQb714m1GCAIR6IsYsAfIy?=
	=?iso-8859-1?q?46xI1CAyOzw+WIQ1IT/iWzn+2nEQ6yBOUK3Mxd+Fb9cIbjgkRsTL8cUTzmnO?=
	=?iso-8859-1?q?EOBwWEWrdSP6YhVV+/YatbcG790M8AyXmwX4MP4HG5LeRJSv7mpxmlpiz1BV?=
	=?iso-8859-1?q?M5W77WLrLi5AhROIVbc5mZEE7xuFP5SS2YE8nyi1rxnGRciItGs3Kk2G/YNL?=
	=?iso-8859-1?q?4+podG7L/v/uQlMZ2U+qFsT+QcJ2VuwSOfXOqLadLyrMgXAiMwtDH8O38FqN?=
	=?iso-8859-1?q?g19wcL0QI5016tEjTJDsFMNVZiPplvkrnze8u7IG60+nSoPnx1vEuSGdh3qt?=
	=?iso-8859-1?q?f2duTckF8gxo+5NhtKURk/6q8HZNReQ/N2XUvdeOVRXbWuJx9POZjhGqoXLA?=
	=?iso-8859-1?q?rKwHbQFBZne/G4PNxpwnt/Wj1z+r6AS9DzXBioA=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 5dc4d55d-bd55-49e4-e06a-08da9040ee20
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:48:54.7095
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 oeGmVoM+8Z40Le8IRvQXACcobzycbM8HR+IwaeojOEm95VkkS1hINo0sGqGhgrABEBtCiUZcKb+HokcNZAsa/w==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SJ0PR10MB5598
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 bulkscore=0
 mlxlogscore=999
 phishscore=0 mlxscore=0 spamscore=0 malwarescore=0 suspectscore=0
 adultscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2207270000 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: BqVgPumVL1CtSbWWbkEQwR8ik72UoVzh
X-Proofpoint-GUID: BqVgPumVL1CtSbWWbkEQwR8ik72UoVzh
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662493784;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=B8vLfVMHYyq9eMWnxQSuV/JdLNr2sKQ59ECjk8k/jxg=;
	b=abbjRMWuTkndwbN3kdKFL4ChBOHMgkATyfjtasrg7OHJen7w/ZtV4hl/rnxUjemPkstFeh
	OoIcRDsmYD4p7gxUXawZ4sjJlaN/Ii5JnSGkIhnpJKp3UEPPw4Hr6/fS+29uKGrDuGORSe
	CvobFcOouh5QNUqJ7niNaiJSnWwFFJY=
ARC-Authentication-Results: i=2;
	imf04.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=cF9Z0SJB;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=ZANBp2h5;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf04.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662493784; a=rsa-sha256;
	cv=pass;
	b=m92OkPXWRhtYd44mrdCSY6syREXkNPZHkMwT0bdeu88M/62+OptFL06zu7AMuUfM0EsX/6
	0jGngccx/1IVv+EdWazGP7ckCBZrYkTWf218J0qsLyODz1B3FpVX42HakajP8kuXyJSium
	qN6Tm4KcBKv3mbp9htxnhrzoBrZaTpA=
X-Rspamd-Queue-Id: 69C7240071
Authentication-Results: imf04.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=cF9Z0SJB;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=ZANBp2h5;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf04.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
X-Rspamd-Server: rspam02
X-Rspam-User: 
X-Stat-Signature: hujwmfqdi3bo1rks7d1tmokaaqrs53yp
X-HE-Tag: 1662493784-487395
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

Use the VMA iterator instead.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
---
 arch/x86/entry/vdso/vma.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/x86/entry/vdso/vma.c b/arch/x86/entry/vdso/vma.c
index 1000d457c332..6292b960037b 100644
--- a/arch/x86/entry/vdso/vma.c
+++ b/arch/x86/entry/vdso/vma.c
@@ -127,17 +127,17 @@ int vdso_join_timens(struct task_struct *task, struct time_namespace *ns)
 {
 	struct mm_struct *mm = task->mm;
 	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	mmap_read_lock(mm);
-
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		unsigned long size = vma->vm_end - vma->vm_start;
 
 		if (vma_is_special_mapping(vma, &vvar_mapping))
 			zap_page_range(vma, vma->vm_start, size);
 	}
-
 	mmap_read_unlock(mm);
+
 	return 0;
 }
 #else
@@ -354,6 +354,7 @@ int map_vdso_once(const struct vdso_image *image, unsigned long addr)
 {
 	struct mm_struct *mm = current->mm;
 	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	mmap_write_lock(mm);
 	/*
@@ -363,7 +364,7 @@ int map_vdso_once(const struct vdso_image *image, unsigned long addr)
 	 * We could search vma near context.vdso, but it's a slowpath,
 	 * so let's explicitly check all VMAs to be completely sure.
 	 */
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		if (vma_is_special_mapping(vma, &vdso_mapping) ||
 				vma_is_special_mapping(vma, &vvar_mapping)) {
 			mmap_write_unlock(mm);

